### PR TITLE
Some code improvements

### PR DESCRIPTION
**Pull request by seregazhuk**
This PR provides some small code improvements. It:
- removes unused variables
- simplifies statements
- removes unnecessary properties initialization

### CODE REVIEW

File: src/Io/PauseBufferStream.php [hunk 1]
```diff
@@ -29,7 +29,7 @@ class PauseBufferStream extends EventEmitter implements ReadableStreamInterface
     private $dataPaused = '';
     private $endPaused = false;
     private $closePaused = false;
-    private $errorPaused = null;
+    private $errorPaused;
```
Comment (clue) [inline src/Io/PauseBufferStream.php]: I understand that these assignments are essentially a NOOP. I've intentionally added these in the past to explicitly state that this property isn't initialized as part of the constructor (see also other occurrences). Would love to hear some more thoughts on this, but won't block this either way. :+1:

**Comment by clue** [comment]
For the reference: This is also somewhat related to #295, does it make sense to cherry-pick (some of) the changes into a single PR?

File: src/Io/PauseBufferStream.php (hunk 1 above)
Comment (seregazhuk) [reply src/Io/PauseBufferStream.php]: As for me, I think that the constructor itself says more explicitly about it, rather then assigned `null` default value. The idea behind these assignments wasn't clear for me, so I've considered them as useless. But if it is a common practice in the code, of course I will undo this change.

**Comment by seregazhuk** [comment]
Sorry, but I'm not sure, that I know how to cherry pick commit from my request to someone's else 😞

**Comment by clue** [comment]
I've just cherry picked @PabloJoan's relevant changes from #295 to this PR :shipit:

**Pull request merged**